### ISSUE TITLE
Enable aggregator routing

### ISSUE THREAD

**Issue by drym3r** [rancher/k3s#120]
**Is your feature request related to a problem? Please describe.**
I'm triying to install [cert-manager](https://github.com/jetstack/cert-manager), but it seems that can't work properly if the kube-apiserver doesn't has the parameter `--enable-aggregator-routing=true`. I understand that this parameter enables the possibility of adding aditional layers to the k8s api.

**Describe the solution you'd like**
Add the parameter `--enable-aggregator-routing=true` to [kube-apiserver](https://github.com/rancher/k3s/blob/828ce5a24a5d7e36a61bcf26167b74ce195e0665/pkg/daemons/control/server.go#L178).

**Describe alternatives you've considered**
I thing is a must if we want to use CDR in k3s, but I'm not sure if there's an alternative.

**Additional context**
The little I know about this I've found it in [this issue](https://github.com/jetstack/cert-manager/issues/1220). From that issue, I found useful [this](https://cert-manager.readthedocs.io/en/master/getting-started/webhook.html#how-it-works) and [this](https://kubernetes.io/docs/tasks/access-kubernetes-api/configure-aggregation-layer/).

**Comment by ibuildthecloud** [comment]
Yes, sorry, this is a know issue that aggregation doesn't work.  I think beyond the args we need to manage another set of certificates.  This is our todo list.

**Comment by drym3r** [comment]
Nice to now it's in your todo! Why do you think you need another set of certificates?

**Comment by adamelliotfields** [comment]
Stumbled upon this now. Sounds like @drym3r and I found all the same links :) 

From my basic understanding, API Aggregation is only required for the Cert Manager [Webhook](https://cert-manager.readthedocs.io/en/latest/getting-started/webhook.html).

The Webhook is only for validating the Cert Manager resources you define (Issuers, ClusterIssuers, Certificates).

I followed the instructions to disable the webhook:

```bash
helm upgrade cert-manager stable/cert-manager --reuse-values --set webhook.enabled=false
```

And after configuring my DNS to point to my server and creating Issuer and ClusterIssuer resources, I was able to install Rancher on k3s via Helm and have a Let's Encrypt staging certificate automatically provisioned (using ingress-nginx instead of Traefik).

As for the additional certs, it says in the aggretation link you posted:

> Warning: Do not reuse a CA that is used in a different context unless you understand the risks and the mechanisms to protect the CA’s usage.
